documentation/concepts.md

```diff
@@ -18,28 +18,57 @@ is then exposed to subscribers. Specifically, Liftbridge centers around the
 of a load-balance group for the stream to join. When there are multiple streams
-in the same group, messages will be balanced among them.
+in the same group, messages will be balanced among them. 
+
+```
+Use case note:
+The usual use case is as follows. A typical subject is the command subject (compare kafka topic) of the CQRS pattern. The corresponding log created by this
+subject is the implementation of the event sourcing pattern. The response of a command being put on a subject is a microservice worker reading that command 
+off the subject and executing the command. Subsequently, the result of this activity is then posted on another subject, perhaps for down streams analytical 
+reporting purposes etc. This enables the Query in the CQRS patterns. A careful reader saw the above remark of replication factor, redundancy and groups. 
+These choices will impact this example with microservice workers, since the order of messages and guarantees on replication will be affected by these chioces. More on that later.
+
+For further info on CQRS and event sourceing pls
+see https://martinfowler.com/bliki/CQRS.html
+and https://martinfowler.com/eaaDev/EventSourcing.html
+respectively. 
+```
 
 There can be multiple streams attached to the same NATS subject, but stream
-names must be unique within a cluster.
+names must be unique within a cluster. 
+
+[@Tyler Please describe if this is intended architecture or just happened due to dependecy towards NATS. If the latter we need to advice on usage. If the former, we need to descibe the use case. ]
+```
+This means that the concept of stream name is now defined as being the least common denominator, not the NATS subject as when using NATS alone without Liftbridge.
+```
 
 By default, streams have a single partition. This partition maps directly to
 the stream's NATS subject. If a stream has multiple partitions, each one maps
 to a different NATS subject derived from the stream subject. For example, if a
 stream with three partitions is attached to the subject "foo", the partitions
-will map to the subjects "foo", "foo.1", and "foo.2", respectively.
+will map to the subjects "foo", "foo.1", and "foo.2", respectively. (Please note the naming convention on these subjects linked to partitions.)
 
 Each partition has its own message log, leader, and set of followers. To reduce
 resource consumption, partitions can be paused. Paused partitions are
 subsequently resumed once they are published to.
 
+```
+Use case note:
+Since each partition has its own message log, it means that to meet a data lineage requirement one could point all these logs to a general log like a Linux system log by subscribing to * or concatenating the logs on the storage device attached. Aggregated or not, these logs could then be consumed by a supervisor system, e.g Logstash, Jaeger, Zipkin etc for DevOps reasons. Or just stored in an S3 bucket for persistence like a ledger of what has been communicated as events.
```

Comment:
This is not for the regular consumer. (Actually this consumer could be treated as a regular consumer but one would probably avoid this) This message log-consumer is present for the Ops team consuming enriched log-events from the liftbridge engine incl. metadata about events depending on flags like -log=INFO etc. They need all logs to be sieved into a log consumer engine for error handling, but also for crime prevention. Since I guess Liftbridge could be used on a non k8s platform this needs to be present somewhere. One analogy could be a bank doing Anti-money laundering applying AI or just Storm/Spark jobs on the regular subject events PLUS these Liftbridge log-events. Does it make sense?